bleak/backends/bluezdbus/client.py

```diff
@@ -769,147 +943,106 @@ async def stop_notify(
-            "StopNotify",
-            interface=defs.GATT_CHARACTERISTIC_INTERFACE,
-            destination=defs.BLUEZ_SERVICE,
-            signature="",
-            body=[],
-            returnSignature="",
-        ).asFuture(asyncio.get_event_loop())
-        self._notification_callbacks.pop(characteristic.path, None)
-
-        self._subscriptions.remove(characteristic.handle)
-
-    # DBUS introspection method for characteristics.
+        if not self.is_connected:
+            raise BleakError("Not connected")
 
-    async def get_all_for_characteristic(
-        self,
-        char_specifier: Union[BleakGATTCharacteristicBlueZDBus, int, str, uuid.UUID],
-    ) -> dict:
-        """Get all properties for a characteristic.
-
-        This method should generally not be needed by end user, since it is a DBus specific method.
-
-        Args:
-            char_specifier: The characteristic to get properties for, specified by either
-                integer handle, UUID or directly by the BleakGATTCharacteristicBlueZDBus
-                object representing it.
-
-        Returns:
-            (dict) Properties dictionary
-
-        """
         if not isinstance(char_specifier, BleakGATTCharacteristicBlueZDBus):
             characteristic = self.services.get_characteristic(char_specifier)
         else:
             characteristic = char_specifier
         if not characteristic:
             raise BleakError("Characteristic {} not found!".format(char_specifier))
 
-        out = await self._bus.callRemote(
-            characteristic.path,
-            "GetAll",
-            interface=defs.PROPERTIES_INTERFACE,
-            destination=defs.BLUEZ_SERVICE,
-            signature="s",
-            body=[defs.GATT_CHARACTERISTIC_INTERFACE],
-            returnSignature="a{sv}",
-        ).asFuture(asyncio.get_event_loop())
-        return out
-
-    async def _get_device_properties(self, interface=defs.DEVICE_INTERFACE) -> dict:
-        """Get properties of the connected device.
-
-        Args:
-            interface: Which DBus interface to get properties on. Defaults to `org.bluez.Device1`.
-
-        Returns:
-            (dict) The properties.
+        reply = await self._bus.call(
+            Message(
+                destination=defs.BLUEZ_SERVICE,
+                path=characteristic.path,
+                interface=defs.GATT_CHARACTERISTIC_INTERFACE,
+                member="StopNotify",
+            )
+        )
+        assert_reply(reply)
 
-        """
-        return await self._bus.callRemote(
-            self._device_path,
-            "GetAll",
-            interface=defs.PROPERTIES_INTERFACE,
-            destination=defs.BLUEZ_SERVICE,
-            signature="s",
-            body=[interface],
-            returnSignature="a{sv}",
-        ).asFuture(asyncio.get_event_loop())
+        self._notification_callbacks.pop(characteristic.path, None)
+        self._subscriptions.pop(characteristic.handle, None)
```

Comment:
`self._subscriptions.pop(characteristic.handle, None)` needs to be `self._subscriptions.pop(characteristic.handle)` since it is a list and not a dict. However, using `characteristic.handle` will yield an `IndexError` here.

Should it be a dcit? Or a set? Or do we apply this:
```python
try:
    self._subscriptions.remove(characteristic.handle)
except ValueError:
    pass
```

Reply:
oops. I'll do the try/except